internal/tiltfile/k8s.go

```diff
@@ -173,7 +200,39 @@ func (s *tiltfileState) filterYaml(thread *starlark.Thread, fn *starlark.Builtin
 	}, nil
 }
 
-func (s *tiltfileState) k8sResource(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+// v1 and v2 resource assembly both have a function named `k8s_resource`. They take different arguments.
+// The goal of `isProbablyV1K8SResourceCall` is to guess if a `k8s_resource` call is still using v1 arguments.
+func (s *tiltfileState) isProbablyV1K8SResourceCall(args starlark.Tuple, kwargs []starlark.Tuple) (result bool, reason string) {
+	// check kwargs
+	var k8sResourceV1OnlyNames = map[string]bool{
+		"name":  true,
+		"yaml":  true,
+		"image": true,
+	}
+	for _, item := range kwargs {
+		name := string(item[0].(starlark.String))
+		if _, ok := k8sResourceV1OnlyNames[name]; ok {
+			return true, fmt.Sprintf("it was called with kwarg %q, which is deprecated", name)
+		}
+	}
+
+	// check positional args
+	// check if the second arg is yaml (v1) instead of a resource name (v2)
+	if args.Len() >= 2 {
+		// `false` because this isn't actually a config file - we just want to test if it's valid k8s yaml and throw it away
+		_, err := s.yamlEntitiesFromSkylarkValueOrList(args[1], false)
```

Comment:
i feel like we could do a much simpler check here? like just check if this is a file with a newline in it?

for example, if the yaml is invalid for some reason (like a missing colon), you still want to print this error message

Reply:
yeah, I guess I was unconsciously trying to avoid false positives, which is a weird goal

also, I think introducing the possibility of false positives means we need an escape hatch for false positives, i.e. don't check this if the user has explicitly opted into v2, so I'll add that